src/Cake/Scripting/Roslyn/RoslynScriptSession.cs

```diff
@@ -78,62 +91,132 @@ public void ImportNamespace(string @namespace)
+
+                    // Create the script options dynamically.
+                    var options = Microsoft.CodeAnalysis.Scripting.ScriptOptions.Default
+                        .AddImports(Namespaces.Except(script.ExcludedNamespaces.Keys))
+                        .AddReferences(References)
+                        .AddReferences(ReferencePaths.Select(r => r.FullPath))
+                        .WithEmitDebugInformation(_options.PerformDebug)
+                        .WithMetadataResolver(Microsoft.CodeAnalysis.Scripting.ScriptMetadataResolver.Default);
+
+                    roslynScript = CSharpScript.Create(code, options, _host.GetType());
+
+                    _log.Verbose("Compiling build script...");
+                    compilation = roslynScript.GetCompilation();
+                    var diagnostics = compilation.GetDiagnostics();
+
+                    var errors = new List<Diagnostic>();
+
+                    foreach (var diagnostic in diagnostics)
+                    {
+                        switch (diagnostic.Severity)
+                        {
+                            case DiagnosticSeverity.Info:
+                                _log.Information(diagnostic.ToString());
+                                break;
+                            case DiagnosticSeverity.Warning:
+                                _log.Warning(diagnostic.ToString());
+                                break;
+                            case DiagnosticSeverity.Error:
+                                _log.Error(diagnostic.ToString());
+                                errors.Add(diagnostic);
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+
+                    if (errors.Any())
+                    {
+                        var errorMessages = string.Join(Environment.NewLine, errors.Select(x => x.ToString()));
+                        var message = string.Format(CultureInfo.InvariantCulture, "Error(s) occurred when compiling build script:{0}{1}", Environment.NewLine, errorMessages);
+                        throw new CakeException(message);
+                    }
+                }
+                if (_scriptCacheEnabled)
+                {
+                    // Verify cache directory exists
+                    if (!IO.Directory.Exists(_scriptCachePath.FullPath))
+                    {
+                        IO.Directory.CreateDirectory(_scriptCachePath.FullPath);
+                    }
+                    if (string.IsNullOrEmpty(scriptHash))
+                    {
+                        scriptHash = FastHash.GenerateHash(Encoding.UTF8.GetBytes(string.Concat(script.Lines)));
+                    }
+                    var emitResult = compilation.Emit(cachedAssembly.FullPath);
+
+                    if (emitResult.Success)
+                    {
+                        IO.File.WriteAllText(hashFile.FullPath, scriptHash);
```

Comment:
Don't use `System.IO` use the `IFileSystem`

Reply:
Fixed